src/experiment/trials/camera.js

```diff
@@ -9,111 +9,105 @@ const WEBCAM_ID = "webcam";
+        // Saves the raw data feed from the participants camera (executed on cameraRecorder.stop()).
+        cameraRecorder.addEventListener("stop", () => {
+          const blob = new Blob(cameraChunks, { type: cameraRecorder.mimeType });
 
-            // Pass video data to Electron as a base64 encoded string
-            const reader = new FileReader();
-            reader.readAsDataURL(blob);
-            reader.onloadend = () => {
-              window.electronAPI.saveVideo(reader.result);
-            };
-          });
-        },
-        on_load: function () {
-          // Assign camera feed to the <video> element
-          const camera = document.getElementById(WEBCAM_ID);
+          // Pass video data to Electron as a base64 encoded string
+          const reader = new FileReader();
+          reader.readAsDataURL(blob);
+          reader.onloadend = () => {
+            window.electronAPI.saveVideo(reader.result);
+          };
+        });
+      },
+      on_load: function () {
+        // Assign camera feed to the <video> element
+        const camera = document.getElementById(WEBCAM_ID);
 
-          camera.srcObject = window.jsPsych.pluginAPI.getCameraRecorder().stream;
-        },
-        on_finish: function () {
-          // Begin video recording
-          window.jsPsych.pluginAPI.getCameraRecorder().start();
-        },
+        camera.srcObject = window.jsPsych.pluginAPI.getCameraRecorder().stream;
+      },
+      on_finish: function () {
+        // Begin video recording
+        window.jsPsych.pluginAPI.getCameraRecorder().start();
       },
-    ],
-  };
-}
+    },
+  ],
+};
 
 /**
  * A trial that finishes recording the participant using their computer's default camera
  *
- * @param {Number} duration How long to show the trial for
  * @returns {Object} A jsPsych trial object
  */
-export function buildCameraEndTrial() {
-  const recordingEndMarkup = h1(LANGUAGE.trials.camera.end);
-
-  return {
-    type: htmlKeyboardResponse,
-    stimulus: div(recordingEndMarkup),
-    trial_duration: 5000,
-    on_start: function () {
-      // Complete the camera recording
+const recordingEndMarkup = h1(LANGUAGE.trials.camera.end);
+export const buildCameraEndTrial = {
+  type: htmlKeyboardResponse,
+  stimulus: div(recordingEndMarkup),
+  trial_duration: 5000,
+  on_start: function () {
+    // Complete the camera recording
 
-      if (!ENV.USE_ELECTRON) {
-        throw new Error("video recording is only available when running inside Electron");
-      }
+    if (!ENV.USE_ELECTRON) {
+      throw new Error("video recording is only available when running inside Electron");
+    }
 
-      const cameraRecorder = window.jsPsych.pluginAPI.getCameraRecorder();
-      if (!cameraRecorder) {
-        console.error("Camera is not initialized, no data will be recorded.");
-        return;
-      }
+    const cameraRecorder = window.jsPsych.pluginAPI.getCameraRecorder();
```

Comment:
I commented on the other out `window.jsPsych`. What's the order of these PRs?

Reply:
Please let me know if this is not what you mean by order, but I think we should merge this branch into `jsPsych-global` and then merge `jsPsych-global` into `feat-v4`